### PR TITLE
Add dynamic 'Uninstall V1' step to Monitoring V2 wizard

### PR DESCRIPTION
**Pull request by richard-cox**
- Add ability to dynamically include additional steps when installing charts
  - Works in a similar way to custom values components (discoverable via component files)
  - Step component will report if it's loaded and if it's hidden
- Add dynamic step to v2 monitoring that is shown if v1 monitoring is installed
  - User is able to uninstall v1 from step
  - step blocks wizard until uninstall occurs

### CODE REVIEW

File: assets/translations/en-us.yaml [hunk 1]
```diff
@@ -1819,6 +1822,13 @@ monitoring:
       repeatInterval: Repeat Interval
   routesAndReceivers: Routes and Receivers
   monitors: Monitors
+  installSteps:
+    uninstallV1:
+      stepTitle: Previous Version
+      stepSubtext: Uninstall Cluster Manager Monitoring
+      warning: Monitoring is currently deployed from Cluster Manager. If you are migrating from an older version of {vendor} with monitoring enabled, please uninstall monitoring before attempting to install the new {vendor} Monitoring chart.
+      success: Cluster Manager monitoring successfully installed. Press Next to continue
```
Comment (richard-cox) [inline assets/translations/en-us.yaml:1830]: stepSubtext, warning & success - `v1 monitoring`

File: components/Wizard.vue [hunk 2]
```diff
@@ -186,106 +208,111 @@ export default {
...
-    <div class="step-container">
-      <template v-for="step in steps">
-        <div v-if="step === activeStep" :key="step.name" class="step-container__step">
-          <slot :step="step" :name="step.name" />
-        </div>
-      </template>
-    </div>
+      <div class="step-container">
+        <template v-for="step in allSteps">
+          <!-- {{ step.name + ' ' + step.hidden + ' ' + step.loading }} -->
```
Comment (richard-cox) [inline components/Wizard.vue:278]: remove debug

File: pages/c/_cluster/apps/install.vue [hunk 3]
```diff
@@ -857,7 +909,7 @@ export default {
   <div v-else class="install-steps">
     <Wizard
       v-if="value"
-      :steps="steps"
+      :all-steps="steps"
```
Comment (richard-cox) [inline pages/c/_cluster/apps/install.vue:912]: change back to steps, tweak inner wizard steps

File: utils/dynamic-importer.js [hunk 4]
```diff
@@ -3,66 +3,74 @@
...
 export function importChart(name) {
-  if ( !name ) {
+  if (!name) {
     throw new Error('Name required');
   }
 
   return () => import(/* webpackChunkName: "chart" */ `@/chart/${name}`);
 }
 
+export function importChartStep(stepPath) {
```
Comment (richard-cox) [inline utils/dynamic-importer.js:37]: test with just importChart

File: chart/monitoring/steps/uninstall-v1.vue [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      {{ t('monitoring.installSteps.uninstallV1.warning') }}
+    </p>
+    <AsyncButton
+      v-if="!uninstalled"
+      mode="uninstall"
+      :delay="2000"
+      :disabled="uninstalled"
+      @click="uninstall"
+    />
+    <p v-else>
```
Comment (richard-cox) [inline chart/monitoring/steps/uninstall-v1.vue:79]: on success.... remove text & button .... use message with icon component on success

**Review by nwmac — APPROVED** [review]

**Pull request merged**